src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.CompilationAnalyzer.cs

```diff
@@ -111,6 +111,9 @@ public override int GetHashCode()
 
                 public override bool Equals(Diagnostic? obj)
                 {
+                    if (obj is CompilerDiagnostic other)
+                        return _original.Equals(other._original);
+
                     return _original.Equals(obj);
```

Comment:
>_original.Equals(obj) [](http://example.com/codeflow?start=27&length=21)

Perhaps component that depends on "wrapped == unwrapped" should use custom comparer that ensure that equality is symmetrical. And this implementation should return false here.

Reply:
Thanks @AlekseyTs. After putting more thought, I agree your suggestion is correct. I have updated the code and also the unit test to match this behavior.